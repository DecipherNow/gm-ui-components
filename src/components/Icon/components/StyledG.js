import styled from "styled-components";
import { PropTypes } from "prop-types";

const StyledG = styled.g.attrs({
  fill: "currentColor"
})`
  transform: scale(${props => props.ratio});
`;

StyledG.propTypes = {
  ratio: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

export default StyledG;
