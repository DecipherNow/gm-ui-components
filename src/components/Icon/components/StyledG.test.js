import React from "react";
import { shallow } from "enzyme";
import StyledG from "./StyledG";

describe("StyledG", () => {
  it("matches snapshot", () => {
    const aStyledG = shallow(<StyledG />);
    expect(aStyledG).toMatchSnapshot();
  });
});
