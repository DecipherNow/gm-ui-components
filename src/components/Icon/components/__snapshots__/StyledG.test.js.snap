// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledG matches snapshot 1`] = `
.c0 {
  -webkit-transform: scale();
  -ms-transform: scale();
  transform: scale();
}

<g
  className="c0"
  fill="currentColor"
/>
`;
